sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala

```diff
@@ -192,14 +193,17 @@ abstract class LogicalPlan extends QueryPlan[LogicalPlan] with Logging {
 
       // One match, but we also need to extract the requested nested field.
       case Seq((a, nestedFields)) =>
-        // The foldLeft adds UnresolvedGetField for every remaining parts of the name,
-        // and aliased it with the last part of the name.
-        // For example, consider name "a.b.c", where "a" is resolved to an existing attribute.
-        // Then this will add UnresolvedGetField("b") and UnresolvedGetField("c"), and alias
-        // the final expression as "c".
-        val fieldExprs = nestedFields.foldLeft(a: Expression)(UnresolvedGetField)
-        val aliasName = nestedFields.last
-        Some(Alias(fieldExprs, aliasName)())
+        // The foldLeft will resolve all of the nested data type, to get its attributes.
+        val fieldExprs = nestedFields.foldLeft(a: Expression) { case (e, fieldName) =>
+          resolveGetField(e, fieldName, resolver)
+        }
+
+        // TODO the alias name is quite tricky to me, set it to _col1, _col2.. ?
+        // Set it as original attribute name like "a.b.c" seems still confusing,
+        // and we may never reference this column by its name (with "."), except
+        // people write SQL like: SELECT a.b.c as newCol FROM nestedTable, which
+        // explicitly specifying the alias name for the output column
+        Some(Alias(fieldExprs, name)())
```

Comment:
I am not so sure how Hive handle that, but it can not be `c`; otherwise it may cause reference arbitrary for its parent logical plan.

e.g. 
Assume we have table `tbl` with schema Struct<a: Struct<b:Int, c: Int>, b: int>
```
SELECT b FROM (SELECT a.b, b FROM tbl)
```